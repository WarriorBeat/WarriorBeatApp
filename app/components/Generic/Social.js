/**
 * Social.js
 * Social Icons
 * components/Generic
 */

import React from "react"
import PropTypes from "prop-types"
import { SocialIcon } from "react-native-elements"
import { Config, Util } from "config"
import { Colors, Position } from "components/styles"

const getProps = (light, hPad) => {
  if (!light) return {}
  return {
    raised: false,
    iconStyle: {
      ...Colors.bgColor("primaryDark"),
      ...Colors.color("whiteSecondary"),
      ...Position.hPadding(hPad),
    },
    light: true,
    iconSize: 42,
  }
}

export const SocialButton = ({
  type, url, onPress, light, hPad,
}) => (
  <SocialIcon
    type={type}
    onPress={() => onPress(url)}
    iconStyle={Position.hPadding(hPad)}
    {...getProps(light, hPad)}
  />
)

const Social = (props) => {
  const socials = Object.entries(Config.social)
  return socials.map(s => <SocialButton key={s[0]} type={s[0]} url={s[1].url} {...props} />)
}

SocialButton.propTypes = {
  type: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  onPress: PropTypes.func,
  light: PropTypes.bool,
  hPad: PropTypes.string,
}

SocialButton.defaultProps = {
  onPress: url => Util.openLink(url),
  light: false,
  hPad: "0%",
}

export default Social
