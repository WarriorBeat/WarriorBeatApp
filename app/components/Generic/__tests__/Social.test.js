/**
 * __tests__/Social.test.js
 * Component Test
 * tests
 */
/* eslint-disable */

import React from "react"
import { shallow, render } from "react-native-testing-library"
import Social, { SocialButton } from "../Social"

test("should render correctly", () => {
  const tree = render(<Social />)
  expect(tree).toMatchSnapshot()
})

test("should have 3 elements", () => {
  const { getAllByType } = render(<Social />)
  const buttons = getAllByType(SocialButton)

  expect(buttons).toHaveLength(3)
})

test("should render lightly", () => {
  const tree = render(<Social light={true} />)
  expect(tree).toMatchSnapshot()
})

test("should render lightly and with 3% padding", () => {
  const tree = render(<Social light={true} hPad="3%" />)
  expect(tree).toMatchSnapshot()
})
