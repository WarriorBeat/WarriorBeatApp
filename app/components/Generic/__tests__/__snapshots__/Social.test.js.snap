// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
Array [
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00aced",
        "borderRadius": 30,
        "flexDirection": "row",
        "height": 52,
        "justifyContent": "center",
        "margin": 7,
        "shadowColor": "rgba(0,0,0, .4)",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
        "width": 52,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 24,
            },
            Object {
              "paddingHorizontal": 0,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#517fa4",
        "borderRadius": 30,
        "flexDirection": "row",
        "height": 52,
        "justifyContent": "center",
        "margin": 7,
        "shadowColor": "rgba(0,0,0, .4)",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
        "width": 52,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 24,
            },
            Object {
              "paddingHorizontal": 0,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#21759b",
        "borderRadius": 30,
        "flexDirection": "row",
        "height": 52,
        "justifyContent": "center",
        "margin": 7,
        "shadowColor": "rgba(0,0,0, .4)",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
        "width": 52,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 24,
            },
            Object {
              "paddingHorizontal": 0,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>,
]
`;

exports[`should render lightly 1`] = `
Array [
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 30,
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 7,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#00aced",
              "fontSize": 42,
            },
            Object {
              "backgroundColor": "#2d2d2d",
              "color": "rgba(255,255,255,0.70)",
              "paddingHorizontal": 0,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 30,
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 7,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#517fa4",
              "fontSize": 42,
            },
            Object {
              "backgroundColor": "#2d2d2d",
              "color": "rgba(255,255,255,0.70)",
              "paddingHorizontal": 0,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 30,
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 7,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#21759b",
              "fontSize": 42,
            },
            Object {
              "backgroundColor": "#2d2d2d",
              "color": "rgba(255,255,255,0.70)",
              "paddingHorizontal": 0,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>,
]
`;

exports[`should render lightly and with 3% padding 1`] = `
Array [
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 30,
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 7,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#00aced",
              "fontSize": 42,
            },
            Object {
              "backgroundColor": "#2d2d2d",
              "color": "rgba(255,255,255,0.70)",
              "paddingHorizontal": 22.5,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 30,
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 7,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#517fa4",
              "fontSize": 42,
            },
            Object {
              "backgroundColor": "#2d2d2d",
              "color": "rgba(255,255,255,0.70)",
              "paddingHorizontal": 22.5,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 30,
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 7,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#21759b",
              "fontSize": 42,
            },
            Object {
              "backgroundColor": "#2d2d2d",
              "color": "rgba(255,255,255,0.70)",
              "paddingHorizontal": 22.5,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>,
]
`;
